msu/hooks/scenarios/world/starting_scenario.nut

```diff
@@ -9,3 +9,22 @@
 	{
 	}
 });
+
+::mods_hookDescendants("scenarios/world/starting_scenario", function(o) {
+	if ("onUpdateLevel" in o)
+	{
+		local onUpdateLevel = o.onUpdateLevel;
+		o.onUpdateLevel = function( _bro )
+		{
+			onUpdateLevel(_bro);
+			_bro.getSkills().onUpdateLevel();
+		}
+	}
+	else
+	{
+		o.onUpdateLevel <- function( _bro )
+		{
+			_bro.getSkills().onUpdateLevel();
+		}
+	}
```

Comment:
This still isn't able to properly handle situations were 2 scenarios inherit from each other, and the one that inherits doesn't have an onUpdateLevel function.

Reply:
I don't understand. If you are playing a scenario that inherits, it will call its parent's onUpdateLevel function which, if it did exist, will have the call on the skills onUpdateLevel. It will work.

Reply:
Check now